wasm_utils/src/lib.rs

```diff
@@ -172,25 +172,23 @@ pub fn deserialize<'s, T: Deserialize<'s>>(ptr_data: *mut c_char) -> T {
 pub fn deserialize_allocation<'s, T: Deserialize<'s>>(encoded_allocation: u32) -> T {
     let allocation = SinglePageAllocation::new(encoded_allocation);
     let allocation = allocation.expect("received error instead of valid encoded allocation");
-    return deserialize(allocation.offset as *mut c_char);
+    unsafe { deserialize(allocation.offset as *mut c_char) }
```

Comment:
For the third time, I would like to keep return statements when it makes the code more readable, like in this case.

Reply:
Since Rust is an expression based language, it is generally frowned upon to use `return` statements. "The Rust programming language" even says so.

Reply:
The "official" notion is that `return` statements should only be used for *early* returns. It took a while for me to get used to this, but eventually this enables you to write more streamlined code.